### PR TITLE
Update missing Mima filters in master for recently released 2.4.x versions

### PR DESCRIPTION
**Pull request by jrudolph**
Forward port of changes in #22311 + some extra cleanup and comments.

Refs #22306 

(Recreated from #22312 which didn't validate for some reason)

### CODE REVIEW

**Comment by patriknw** [comment]
The mima issues are:
```
[error]  * method EnableAdditionalSerializationBindings()Boolean in class akka.actor.ActorSystem#Settings does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("akka.actor.ActorSystem#Settings.EnableAdditionalSerializationBindings")
[error]  * method AllowJavaSerialization()Boolean in class akka.actor.ActorSystem#Settings does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("akka.actor.ActorSystem#Settings.AllowJavaSerialization")
[error]  * the type hierarchy of class akka.actor.Terminated is different in current version. Missing types {akka.actor.NoSerializationVerificationNeeded}
[error]    filter with: ProblemFilters.exclude[MissingTypesProblem]("akka.actor.Terminated")
[error]  * class akka.serialization.DisabledJavaSerializer#JavaSerializationException does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("akka.serialization.DisabledJavaSerializer$JavaSerializationException")
[error]  * object akka.serialization.BaseSerializer does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("akka.serialization.BaseSerializer$")
[error]  * method AllowJavaSerialization()Boolean in class akka.serialization.Serialization does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("akka.serialization.Serialization.AllowJavaSerialization")
[error]  * object akka.serialization.DisabledJavaSerializer does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("akka.serialization.DisabledJavaSerializer$")
[error]  * class akka.serialization.DisabledJavaSerializer does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("akka.serialization.DisabledJavaSerializer")
```

I guess that will be fixed when we fw port that serialization PR?

**Comment by jrudolph** [comment]
Yes,  exactly. We should rebuild once the forward port was merged to
verify.

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
New MiMa failures were introduced by #22320. Added a commit to fix them, now waiting for verification to be merged finally.

File: project/MiMa.scala
Comment (ktoso) [inline project/MiMa.scala]: Good

**Review by ktoso — APPROVED** [review]
Lgtm👌

**Comment by jrudolph** [comment]
Thanks, @ktoso.

**Pull request merged**
